File: public/docs/_examples/template-syntax/ts/src/app/my-click.directive.ts
import {Directive,  Output, ElementRef, EventEmitter} from 'angular2/core';

@Directive({selector:'[my-click]'})
export class MyClickDirective {
  // #docregion my-click-output-1
  @Output('myClick') clicks = new EventEmitter<string>();
 // #enddocregion my-click-output-1
  constructor(el: ElementRef){
    el.nativeElement
      .addEventListener('click', (event:Event) => {
        this.clicks.emit('Click!');
      });
  }
}

// #docregion my-click-output-2
@Directive({
// #enddocregion my-click-output-2
  selector:'[my-click2]',
// #docregion my-click-output-2
  outputs:['clicks:myClick']
})
// #enddocregion my-click-output-2
export class MyClickDirective2 {
  clicks = new EventEmitter<string>();
  constructor(el: ElementRef){
    el.nativeElement
      .addEventListener('click', (event:Event) => {
        this.clicks.emit('Click!');
      });
  }
}

File: public/docs/_examples/template-syntax/dart/lib/my_click_directive.dart
// #docplaster
import 'dart:html';

import 'package:angular2/core.dart';

@Directive(selector: '[myClick]')
class MyClickDirective {
  // #docregion my-click-output-1
  // @Output(alias) [type info] propertyName = ...
  @Output('myClick') final EventEmitter clicks = new EventEmitter<String>();

  // #enddocregion my-click-output-1
  bool _toggle = false;

  MyClickDirective(ElementRef el) {
    Element nativeEl = el.nativeElement;
    nativeEl.onClick.listen((Event e) {
      _toggle = !_toggle;
      clicks.emit(_toggle ? 'Click!' : '');
    });
  }
}

// #docregion my-click-output-2
@Directive(
// #enddocregion my-click-output-2
    selector: '[myClick2]',
// #docregion my-click-output-2
    // ...
    outputs: const ['clicks:myClick']) // propertyName:alias
// #enddocregion my-click-output-2
class MyClickDirective2 {
  final EventEmitter clicks = new EventEmitter<String>();
  bool _toggle = false;

  MyClickDirective2(ElementRef el) {
    el.nativeElement.onClick.listen((Event e) {
      _toggle = !_toggle;
      clicks.emit(_toggle ? 'Click2!' : '');
    });
  }
}
